src/main/scala/User.scala

```diff
@@ -1,3 +1,108 @@
-object User {
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import spray.json._
 
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+/*
+  POST /register -> register User(username, password)
+  GET /users -> get all the users
+  GET /user/{id} -> get a user by id
+  GET /login -> OK (Successful), NotFound (Error)
+ */
+
+case class UserInfo(username: String, password: String)
+
+trait UserJsonProtocol extends DefaultJsonProtocol{
+  implicit val userFormat = jsonFormat2(UserInfo)
+}
+
+object User extends App with UserJsonProtocol{
+
+  implicit val system = ActorSystem("User")
+  import system.dispatcher
+
+  var users = Map[Int, UserInfo]()
+  var id: Int = 0
+
+  def findUser(username: String, password: String): Boolean = {
+    if(!(users.values.toList.find(i => i.username == username && i.password == registerHashPass(password)).isEmpty)) true
+    else false
+  }
+
+  def findUserById(id: Int): Option[UserInfo] ={
+    users.get(id)
+  }
+  def registerHashPass(password: String): String = password.hashCode.toString
+
+  val registerRoute =
+    (path("register") & post & extractRequest & extractLog){ (request, log) =>
+      val entity = request.entity
+      val strictEntity = entity.toStrict(2 seconds)
```

Comment:
I think this is blocking code. You should try to use a `.map` or a `for` instead